pulsar/consumer_partition.go

```diff
@@ -402,23 +404,27 @@ func (pc *partitionConsumer) MessageReceived(response *pb.CommandMessage, header
 		var msg *message
 		if smm != nil {
 			msg = &message{
-				publishTime: timeFromUnixTimestampMillis(msgMeta.GetPublishTime()),
-				eventTime:   timeFromUnixTimestampMillis(smm.GetEventTime()),
-				key:         smm.GetPartitionKey(),
-				properties:  internal.ConvertToStringMap(smm.GetProperties()),
-				topic:       pc.topic,
-				msgID:       msgID,
-				payLoad:     payload,
+				publishTime:         timeFromUnixTimestampMillis(msgMeta.GetPublishTime()),
+				eventTime:           timeFromUnixTimestampMillis(smm.GetEventTime()),
+				key:                 smm.GetPartitionKey(),
+				properties:          internal.ConvertToStringMap(smm.GetProperties()),
+				topic:               pc.topic,
+				msgID:               msgID,
+				payLoad:             payload,
+				replicationClusters: msgMeta.GetReplicateTo(),
```

Comment:
why is this added?

Reply:
We need to copy all properties of the original message and noticed that this was not being exposed in the internal struct.